app/services/twilio_service.rb

```diff
@@ -35,6 +35,21 @@ def from_number
     "+1#{account['number']}"
   end
 
+  def valid_number?(to_number)
+    to_number = format_phone_number(to_number)
+    begin
```

Comment:
I think we can simplify the logic in this a ton if we use the implicit top-level rescue:

```ruby
def valid_number?(raw_to_number)
  to_number = format_phone_number(raw_to_number)
  response = client.lookups.phone_numbers(to_number).fetch
  response.phone_number
  true
rescue Twilio::SomeSpecificErrorClass => err # TODO: actual error class
  raise unless e.code == TWILIO_BAD_NUMBER_CODE # TODO: better constant name
  false
end
```